README.md

```diff
@@ -24,15 +23,77 @@ module.exports = ({ env }) => ({
       secretAccessKey: env('AWS_ACCESS_SECRET'),
       region: env('AWS_REGION'),
       params: {
-        Bucket: env('AWS_BUCKET'),
+        bucket: env('AWS_BUCKET'),
       },
       baseUrl: env('CDN_BASE_URL'), // e.g. https://cdn.example.com, this is stored in strapi's database to point to the file
-      prefix: env('BUCKET_PREFIX') // e.g. strapi-assets, note the missing slash at the start
+      prefix: env('BUCKET_PREFIX'), // e.g. strapi-assets, note the missing slash at the start
+    },
+  },
+  // ...
+});
+```
+
+### Strapi v4
+
+If using >4.0.0, please use the below config:
+
+`./config/plugins.js`
+```js
+module.exports = ({ env }) => ({
+  // ...
+  upload: {
+    config: {
+      provider: 'strapi-provider-upload-aws-s3-advanced',
+      providerOptions: {
+        accessKeyId: env('AWS_ACCESS_KEY_ID'),
+        secretAccessKey: env('AWS_ACCESS_SECRET'),
+        region: env('AWS_REGION'),
+        params: {
+          bucket: env('AWS_BUCKET'),
+        },
+        baseUrl: env('CDN_BASE_URL'), // e.g. https://cdn.example.com, this is stored in strapi's database to point to the file
+        prefix: env('BUCKET_PREFIX'), // e.g. strapi-assets, note the missing slash at the start
+      },
     },
   },
   // ...
 });
 ```
+#### Image Previews
+
+To allow the thumbnails to properly populate, add the below config to
+
+`./config/plugins.js`
+```js
+module.exports = ({env}) => ([
+  // ...
+  {
+    name: 'strapi::security',
+    config: {
+      contentSecurityPolicy: {
+        useDefaults: true,
+        directives: {
+          'connect-src': ["'self'", 'https:'],
+          'img-src': [
+            "'self'",
+            'data:',
+            'blob:',
+            `${env('CF_BASE_URL')}`
```

Comment:
```suggestion
            `${env('CDN_BASE_URL')}`
```
The config above uses the `CDN_BASE_URL` env variable as reference to a Cloudfront endpoint, I imagine `CF_BASE_URL` is an additional one created by yourself? Could you maybe converge this?

Reply:
You're right about that - I did leave those in there from my own Cloudfront config. Fixed here.